### PR TITLE
Introduce and use decorators for graph sanitizers

### PR DESCRIPTION
**Pull request by sdmccabe**
This PR (in addition to a few cosmetic changes like cleaning up typos I
encountered) introduces two decorators, `@unweighted` and `@undirected`, that
apply the `ensure_unweighted()` and `ensure_undirected()` functions to each
positional argument to a function that is subclassed from `nx.Graph()`.

### CODE REVIEW

File: netrd/distance/quantum_jsd.py [hunk 1]
```diff
@@ -20,14 +20,16 @@
...
 from scipy.linalg import expm
 from .base import BaseDistance
-from ..utilities import ensure_undirected, ensure_unweighted
+from ..utilities import undirected, unweighted
 
 
 class QuantumJSD(BaseDistance):
     """Compares the spectral entropies of the density matrices."""
 
+    @undirected
```
Comment (leotrs) [inline netrd/distance/quantum_jsd.py]: Does `wraps` work well when chaining decorators like this?

File: netrd/distance/onion_divergence.py [hunk 2]
```diff
@@ -59,10 +61,8 @@ def dist(self, G1, G2, dist='lccm'):
 
         """
         # take the simple graph version
-        G1_simple = ensure_undirected(G1)
-        G2_simple = ensure_undirected(G2)
-        G1_simple = ensure_unweighted(G1_simple)
-        G2_simple = ensure_unweighted(G2_simple)
+        G1_simple = G1
```
Comment (leotrs) [inline netrd/distance/onion_divergence.py]: Do we need to rename these args?

File: netrd/distance/netsimile.py [hunk 3]
```diff
@@ -12,21 +12,22 @@
...
 class NetSimile(BaseDistance):
     """Compares node signature distributions."""
 
+    @undirected
+    @unweighted
     def dist(self, G1, G2):
         """A scalable approach to network similarity.
 
-        A network similarity measure based on node signature distrubtions.
+        A network similarity measure based on node signature distributions.
```
Comment (leotrs) [inline netrd/distance/netsimile.py]: distrubtions was funny

File: netrd/distance/onion_divergence.py (hunk 2 above)
Comment (sdmccabe) [reply netrd/distance/onion_divergence.py]: I think I would prefer keeping the `G1_simple` names and making them copies of `G`. I never have a great feel for when the underlying graph gets modified, but I think right now we'll kill the self-loops in the existing graph after calling `dist`.

File: netrd/distance/quantum_jsd.py (hunk 1 above)
Comment (sdmccabe) [reply netrd/distance/quantum_jsd.py]: I think it does, but I'll admit I didn't check that. Will report back later.

**Comment by sdmccabe** [comment]
> The code LGTM, but is there any sensible way of adding meaningful tests here?

We have tests for handling weighted and directed input. Given that, the fact that the existing tests passed suggest that the decorators work, at least on the graph distance side. The decorators are also used in dynamics, where test coverage is considerably worse.

Not systematic, but if you comment out `@unweighted` on `dkSeries` or `@undirected` on `QuantumJSD`, the relevant tests fail.


> Also, should there be Warnings when converting? I don't like changing input data silently.

The decorators are calling the `ensure_undirected()` and `ensure_unweighted()` functions, which emit Warnings.

File: netrd/distance/netsimile.py (hunk 3 above)
Comment (sdmccabe) [reply netrd/distance/netsimile.py]: We put it off in the past, but given that this PR touches a ton of files we should just bite the bullet and do a general cleanup of the docs. I avoided modifying irrelevant files in this PR but there are still a bunch of typos in the docs, and some inconsistency in whether helper functions are documented.

File: netrd/distance/quantum_jsd.py (hunk 1 above)
Comment (sdmccabe) [reply netrd/distance/quantum_jsd.py]: A lazy update on this: [this should be the purpose of using `wraps()`](https://stackoverflow.com/questions/739654/how-to-make-a-chain-of-function-decorators).

**Pull request merged**